endpoints/openrtb2/amp_auction.go

```diff
@@ -262,6 +263,21 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	}
 }
 
+// Sets the effective publisher ID for an AMP request
+func setEffectiveAmpPubID(req *openrtb.BidRequest, urlValues url.Values) {
+	// Try to look for the pub ID in the ususal places first
+	setEffectivePubID(req)
+
+	// For amp requests, the publisher ID could be sent via the account
+	// query string
+	if req.Site.Publisher.ID == pbsmetrics.PublisherUnknown {
+		if acc := urlValues.Get("account"); acc != "" {
```

Comment:
There is already a community implementation in `overrideWithParams`.

Reply:
Sorry, I missed that change as it was done before I started working on this. I will remove the duplicate code and update the PR. Thanks for pointing it out :)